test/runtests.jl

```diff
@@ -41,26 +43,38 @@ using Test
         end
     end
 
-    @testset "@cached_call" begin
+    @testset "@cached_call and @hash_call" begin
         @testset "f()" begin
             f() = 2
-            @test f() == @cached_call(f()) == @cached_call f() # test first and second call
+            call1 = @cached_call f()
+            call2 = @cached_call f()
+            @test f() == call1 == call2
+
+            @test @hash_call(f()) isa UInt
         end
 
         @testset "f(a, b)" begin
             f(a, b) = a + b
-            @test f(1, 2) == @cached_call f(1, 2)
-            @test f(1, 2) != @cached_call f(2, 2)
+            a = 1
```

Comment:
test whether args can be passed as variables not just literals (they were allowed before but not tested)